### ISSUE TITLE
[JENKINS-58608] Add Ability to See Updates and Security Warnings for All Plugins or Just Requested Plugins

### ISSUE THREAD

**Issue by hp77-creator** [jenkinsci/plugin-installation-manager-tool#267]
Users might want to be able to specify if they want to see available updates or security warnings for only the plugins they specified, or if this should also include installed plugins.

Reporter: Natasha Stopa

#202

**Comment by AmanGupta677** [comment]
can you please explain this issue I want to work on this issue.

**Comment by MarkEWaite** [comment]
I believe it is suggesting that a feature might be added to use the contents of the existing plugins directory as the definition of the plugins that should be included in the check for security warnings.  I think that is what is meant by "installed plugins".

I think that "plugins they specified" might mean only the plugins that are listed on the command line or in the plugins.txt file, without considering dependencies and without considering other plugins that may already exist in the plugins directory.

**Comment by jiakuanghe** [comment]
So far, there is already an option called `--available-updates` to see the updates. By default, the security warnings will be shown now. Users can use `--plugins` option to specify which plugins they want to install. I am wondering if there is still anything I need to do about the issue? @MarkEWaite @hp77-creator

**Comment by MarkEWaite** [comment]
> I am wondering if there is still anything I need to do about the issue?

Yes, there is more to do, at least based on my understanding of the request.

When the available updates and security warnings are displayed today, they are displayed for the plugins specified by the user and for all the dependencies of those plugins.  I interpret the request to mean that they want an option that will display available updates and security warnings for only the plugins specified by the user.

The plugins specified by the user are given on the command line or in plugins.txt or in a yaml file.  The dependencies of those plugins are computed from the plugins specified by the user and may include many plugins that are not specified by the user.  

For example, the [Docker installation instructions](https://www.jenkins.io/doc/book/installing/docker/) install plugins that will be used in tutorials with the following command:

```
RUN jenkins-plugin-cli --plugins "blueocean docker-workflow"
```

In that example, the user specified plugins are "blueocean" and "docker-workflow".  Updates and security warnings with this new command line argument would only be reported for those two plugins, not for the many other plugins that are installed as dependencies of those two plugins.
